src/main/java/org/springframework/data/jpa/repository/support/CrudMethodMetadata.java

```diff
@@ -48,20 +51,56 @@
 	 * Returns all query hints to be applied to queries executed for the CRUD method.
 	 *
 	 * @return
+	 * @deprecated use {@link #getQueryHintList()} instead.
 	 */
+	@Deprecated
 	Map<String, Object> getQueryHints();
```

Comment:
It should be fine to return `QueryHints` as this type is store-internal API that shouldn't be called externally.